plugins/services/src/js/events/__tests__/SDKEndpointsActions-test.js

```diff
@@ -0,0 +1,137 @@
+const RequestUtil = require("mesosphere-shared-reactjs").RequestUtil;
+
+const Config = require("#SRC/js/config/Config");
+const ActionTypes = require("../../constants/ActionTypes");
+const AppDispatcher = require("#SRC/js/events/AppDispatcher");
+const SDKEndpointsActions = require("../SDKEndpointActions");
+
+describe("SDKEndpointsActions", function() {
+  describe("#fetchEndpoints", function() {
+    const serviceId = "foo";
+
+    context("#RequestUtil", function() {
+      beforeEach(function() {
+        spyOn(RequestUtil, "json");
+        SDKEndpointsActions.fetchEndpoints(serviceId);
+        this.configuration = RequestUtil.json.calls.mostRecent().args[0];
+      });
+
+      it("calls #json from the RequestUtil", function() {
+        expect(RequestUtil.json).toHaveBeenCalled();
+      });
+
+      it("sends data to the correct URL", function() {
+        expect(this.configuration.url).toEqual(
+          `${Config.rootUrl}/service/${serviceId}/v1/endpoints`
+        );
+      });
+
+      it("uses GET for the request method", function() {
+        expect(this.configuration.method).toEqual("GET");
+      });
+
+      it("dispatches the correct action when successful", function() {
```

Comment:
Please make sure to test that all expected properties are emitted on both success and error - not just `type`. Sorry I missed this on the first review 😞

Reply:
Yes, listening now to all properties.